### PR TITLE
Etcd

### PR DESCRIPTION
**Pull request by Ranjandas**
Added etcd role

```
* Added etcd role, we will evaluate etcd as kv store for docker multihost
networking and docker swarm clustering.
* Added Group Vars for vagrant and default all.yml
* Modified site.yml to accomodate etcd
* Modified vagrant inventory to add new group

This is a WIP. More work has to be done for tyding up the roles.
```

### CODE REVIEW

File: ansible/group_vars/vagrant.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+
+# Various Network Interfaces on the Servers
+priv_interface: ansible_eth1
```
Comment (sebinjohn) [inline ansible/group_vars/vagrant.yml]: what is `priv`?

File: ansible/roles/etcd/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+galaxy_info:
+  author: Ranjandas A P
+  description: Etcd Role for Upswing
+  company: DigitalEarn (Infrastructure)
```
Comment (sebinjohn) [inline ansible/roles/etcd/meta/main.yml]: DigitalEarns??

File: ansible/roles/etcd/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+# tasks file for etcd
+
+- name: install etcd from centos extras repository
+  yum: name=etcd state=installed
+
+# TODO
```
Comment (sebinjohn) [inline ansible/roles/etcd/tasks/main.yml]: TODO needed?

File: ansible/roles/etcd/templates/etcd.conf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+ETCD_NAME={{ ansible_hostname | default('default') }}
+ETCD_DATA_DIR="{{ etcd_data_dir | default('/var/lib/etcd/default.etcd') }}"
+
+
+ETCD_INITIAL_ADVERTISE_PEER_URLS="http://{{ hostvars[inventory_hostname][priv_interface].ipv4.address }}:2380"
+ETCD_INITIAL_CLUSTER="{% for host in groups['etcd_masters']%}{{ hostvars[host]['ansible_hostname']}}=http://{{ hostvars[host][priv_interface]['ipv4']['address'] }}:2380{% if not loop.last%},{% endif %}{% endfor %}"
```
Comment (sebinjohn) [inline ansible/roles/etcd/templates/etcd.conf]: its good to put a space before `%}` in the `{% for..` and `['ansible_hostname']` `loop.last`

File: ansible/roles/etcd/templates/etcd.conf [hunk 5]
```diff
@@ -0,0 +1,12 @@
+ETCD_NAME={{ ansible_hostname | default('default') }}
```
Comment (sebinjohn) [inline ansible/roles/etcd/templates/etcd.conf:1]: Should we make the port numbers also different variables?

File: ansible/group_vars/vagrant.yml (hunk 1 above)
Comment (Ranjandas) [inline ansible/group_vars/vagrant.yml]: `priv` represents Private Interface.

File: ansible/roles/etcd/meta/main.yml (hunk 2 above)
Comment (Ranjandas) [inline ansible/roles/etcd/meta/main.yml]: Yea , typo will fix.

File: ansible/roles/etcd/templates/etcd.conf (hunk 4 above)
Comment (Ranjandas) [inline ansible/roles/etcd/templates/etcd.conf]: Thanks, will fix.

File: ansible/roles/etcd/templates/etcd.conf (hunk 5 above)
Comment (Ranjandas) [inline ansible/roles/etcd/templates/etcd.conf:1]: Yes, we can. Some more improvements planned. will update.

File: ansible/roles/etcd/tasks/main.yml (hunk 3 above)
Comment (Ranjandas) [inline ansible/roles/etcd/tasks/main.yml]: TODO not needed. Removed. Thanks

File: ansible/group_vars/vagrant.yml (hunk 1 above)
Comment (sebinjohn) [inline ansible/group_vars/vagrant.yml]: What you think about changing the variable name to `private_network_interface`?

File: ansible/roles/etcd/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+# tasks file for etcd
+
+- name: install etcd from centos extras repository
+  yum: name=etcd state=installed
```
Comment (sebinjohn) [inline ansible/roles/etcd/tasks/main.yml]: what about when a newer version releases? `state=installed` will not update the packages

Comment (Ranjandas) [inline ansible/roles/etcd/tasks/main.yml]: updated to `state=latest`. Not sure how will this be when there is a major version change with features breaking backward compatibility.

Comment (sebinjohn) [inline ansible/roles/etcd/tasks/main.yml]: we could specify the version to be installed, to ensure everyone gets the same version?

**Pull request merged**